### PR TITLE
CBL-623: Merge flags fix

### PR DESCRIPTION
**Pull request by bmeike**

### CODE REVIEW

**Comment by pasin** [comment]
Can you add the unit tests to ensure that the flags get carried?

**Comment by bmeike** [comment]
@Pasin: yeah.  Good point.

**Comment by bmeike** [comment]
Honestly, I have no idea what the checks are on about.  Builds and all tests pass on my machine.

**Comment by bmeike** [comment]
The test is over here: https://github.com/couchbaselabs/couchbase-lite-java-ee/pull/17/files

File: lib/src/shared/main/java/com/couchbase/lite/internal/core/C4Document.java [hunk 1]
```diff
@@ -88,6 +88,8 @@ public FLDict getSelectedBody2() {
 
     // - Lifecycle
 
+    public int getSelectedFlags() { return withHandle(C4Document::getSelectedFlags, 0); }
```
Comment (jayahariv) [inline lib/src/shared/main/java/com/couchbase/lite/internal/core/C4Document.java]: assuming this is selected _revision_ flag, and not document flag.

Comment (bmeike) [reply lib/src/shared/main/java/com/couchbase/lite/internal/core/C4Document.java]: Yeah... I followed it into the JNI.  It is the selected flag.

**Pull request merged**